addons/web/static/src/core/popover/popover.js

```diff
@@ -11,7 +11,19 @@ export class Popover extends Component {
             position: this.props.position,
         });
     }
-    onPositioned(el, { direction, variant }) {
+    onPositioned(el, { top, left, direction, variant }) {
+        const arrowEl = el.firstElementChild;
+        
+        // Modify positioning if the popup has an arrow
+        const arrowBox = arrowEl.style.display != "none" && arrowEl.getBoundingClientRect();
+        if (arrowBox && direction === "left") {
+            left -= arrowBox.height;
+            el.style.left = `${left}px`;
+        } else if (arrowBox && direction === "top") {
+            top -= arrowBox.height;
+            el.style.top = `${top}px`;
+        }
```

Comment:
```suggestion
        if (arrowBox) {
            if (direction === "left") {
                left -= arrowBox.height;
                el.style.left = `${left}px`;
            } else if (direction === "top") {
                top -= arrowBox.height;
                el.style.top = `${top}px`;
            }
        }
```
just a little rework :slightly_smiling_face:

Reply:
Out of curiosity, why isn't needed for right and bottom?